python/pyspark/pandas/data_type_ops/base.py

```diff
@@ -219,6 +219,10 @@ def _is_boolean_type(right: Any) -> bool:
     )
 
 
+def _is_extension_dtypes(object: Any) -> bool:
```

Comment:
can you explain the fix please?

Reply:
Yeah, basically this function just simply checks whether the given `object` is an extension dtype or not.

Looking at the changes in the `num_ops.py`, it is necessary to handle binary operations differently when either the `left` or `right` is an extension dtype. If this util function doesn't exist, the code:

```python
if isinstance(getattr(left, "dtype", None), extension_dtypes) or isinstance(getattr(right, "dtype", None), extension_dtypes):
    ...
```

would have to be added to every function.

So I added this utility function to simplify the code because I believe that adding such code to every function would compromise the code's readability.